OA-PASS/pass-docker
# Add external ActiveMQ broker

## Overview
* Adds a new `activemq` image to serve as an external broker to Fedora
* Adds a new `pass-jms-external.xml` config file to `fcrepo` that doesn't launch an internal message broker and instead connects to an external one
* Configures `fcrepo`, `indexer`, `deposit` to connect to the external broker.
* Changes the name of the queue the indexer uses from `fedora` to `indexer`

## To Test
* Check out this PR, do `docker-compose pull` (all images are pushed)
* Start the full stack with `docker-compose up`
* Log into Ember as one of the test users (e.g. `incomplete-nih-user:moo`)
* Log into the activemq management console `http://localhost:8161`, user/pass `admin:admin`), see if you can see connected clients and queue contents, does it look right?
  * Note, logging in creates one update to a `User` resource.  You should see one message pass through the `indexer` queue inherent from logging in and having user info updated based on shib attributes

### For deeper testing:
* Create and submit a submission.  Look at activemq console to verify that messages have been enqueued and consumed by deposit services
* Stop (`docker-compose kill`) and start (`docker-compose up`) `fcrepo`, `activemq`, `indexer`, and `deposit` in various permutations to see if the various services can re-connect and recover.

## Open Questions
* ~Should we configure a username/pass for activemq connectivity?~ Not yet, wait for answer from Derek
* ~Should we proxy the activemq management console (and restrict to specific shib users?)~ No

## TODO
* Update/create READMEs
* ~More testing of failure scenarios~ Karen tested quite a few
* ~Security config, if necessary~. Not now, wait for answer from Derek and make any changes in new PR
